### PR TITLE
Fix ElectronContainer getMainWindow()

### LINKED ISSUE #87: ElectronContainer getMainWindow should return first window
Change main window concept of an Electron application to be the first window created.  If it doesn't exist or is closed then return undefined.

### PR DESCRIPTION
**Pull request by bingenito**
Fixes #87

### CODE REVIEW

**Comment by bingenito** [comment]
We have to pretend there is a concept of a main window in Electron apps by simply returning the first window that is created.

**Comment by psmulovics** [comment]
What if the first window was a splash or a login window...?

**Comment by bingenito** [comment]
@psmulovics I'm thinking of removing getMainWindow() completely now and expect user to always use getWindowBy*.  For browser it doesn't work (#42) either.

**Comment by psmulovics** [comment]
You can softly demand marking a window the mainwindow somehow?

**Comment by bingenito** [comment]
I'm starting to think the concept of a main window shouldn't even be assumed at the framework level.  It only applies to OpenFin and even there it can be an issue as some apps have an initial main window that is never visible.

**Comment by bingenito** [comment]
As discussed, will provide an api for dev to register a main window and add lifecycle activity events and other features through this registration.  A good use case of this addition is to be able to quit an Electron app when the "main" window is closed.

File: src/Electron/electron.ts [hunk 1]
```diff
@@ -262,12 +261,14 @@ export class ElectronContainer extends WebContainerBase {
...
-            const win = this.browserWindow.fromId(1);
-            this.mainWindow = win ? this.wrapWindow(win) : undefined;
+        for (const window of this.browserWindow.getAllWindows()) {
+            if (window.options && window.options.main) {
+                return this.wrapWindow(window);
+            }
         }
 
-        return this.mainWindow;
+        const win = this.browserWindow.fromId(1);
```
[the remark this answers is not in the document]
Comment (bingenito) [reply src/Electron/electron.ts]: Will return undefined

[the remark this answers is not in the document]
Comment (bingenito) [reply src/Electron/electron.ts]: Let consumer handle how they want

**Comment by bingenito** [comment]
I want to try and match this behavior on OpenFin and the browser too.  For OpenFin, allow you to specify a different window as main if your current main window is always hidden.

File: src/Electron/electron.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bingenito) [reply src/Electron/electron.ts]: I think I misunderstood your question - if none are marked as main, fallback to first window which matches OpenFin concept.  The main option allows you to override.

[the remark this answers is not in the document]
Comment (bingenito) [reply src/Electron/electron.ts]: Just added a comment to the code

**Pull request merged**